src/ContentClient.ts

```diff
@@ -235,11 +238,63 @@ export class ContentClient implements ContentAPI {
       queryParams.set('fields', [fieldsValue])
     }
 
-    // Reserve a few chars to send the offset
-    const reservedChars = `&offset=`.length + ContentClient.CHARS_LEFT_FOR_OFFSET
+    let reservedChars: number
+    let modifyQueryBasedOnResult: (result: PartialDeploymentHistory<T>, builder: QueryBuilder) => void
+
+    const canWeUseLocalTimestampInsteadOfOffset =
+      deploymentOptions?.fields &&
+      deploymentOptions.fields.isFieldIncluded('auditInfo') &&
+      (!deploymentOptions?.sortBy || deploymentOptions.sortBy.field === SortingField.LOCAL_TIMESTAMP)
+
+    if (canWeUseLocalTimestampInsteadOfOffset) {
+      // Note: the approach used below will get stuck if all 500 deployments on the same page have the same localTimestamp, but that is extremely unlikely
+      reservedChars = '&fromLocalTimestamp='.length + 13
```

Comment:
I haven't read the rest of the code but... WTF?

Reply:
I know it looks weird, but there is a reason for it. We need to 'reserve' space in the url query for the `fromLocalTimestamp` query param. Otherwise, it won't fit